### PR TITLE
feat(frontend): getBtcBalance function

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to create a function for fetching BTC balance.

### CODE REVIEW

File: src/frontend/src/lib/api/signer.api.ts [hunk 1]
```diff
@@ -14,6 +14,17 @@ export const getBtcAddress = async ({
...
 };
 
+export const getBtcBalance = async ({
+	identity,
+	network
+}: {
+	identity: OptionIdentity;
+	network: BitcoinNetwork;
+}): Promise<bigint> => {
+	const { caller_btc_balance } = await getSignerActor({ identity });
```
[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/api/signer.api.ts:24]: If I understood correctly, you have doubts on how I'm gonna implement `BitcoinListener`, right? If yes, I don't know the answer yet. For now, I was going to implement a new service for BTC as well as to create a new loader component that will be used in `Loaders`.

[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/lib/api/signer.api.ts:24]: The balance and transactions should be fetched from a web worker. I doubt `getSignerActor` can be called and used on the web worker side.

[the remark this answers is not in the document]
Comment (lmuntaner) [reply src/frontend/src/lib/api/signer.api.ts:24]: What is the alternative then?

[the remark this answers is not in the document]
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/api/signer.api.ts:24]: [For loading ETH un-certified balances](https://github.com/dfinity/oisy-wallet/blob/main/src/frontend/src/eth/services/balance.services.ts#L26), I don't see a web worker involved (unless its implementation is hidden behind the `ethers` code). Can you please explain why for BTC it should be different?

[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/lib/api/signer.api.ts:24]: > Can you please explain why for BTC it should be different?

IC = web workers

> What is the alternative then?

Pick any other `api` functions called from a web worker as example. You need to get the agent and create the canister. [For example `icrc-index-ng.api.ts`](https://github.com/dfinity/oisy-wallet/blob/4c5445d1b937ada2a9dfe924a0d5d081c6659503/src/frontend/src/icp/api/icrc-index-ng.api.ts#L52)

One of the important part, beside not using i18n store is passing the canister ID given that constant are inexistant in the worker.

**Comment by DenysKarmazynDFINITY** [comment]
Closing this PR as it's not supposed to be merged in the current state (implementation of getting BTC balance via a worker - https://github.com/dfinity/oisy-wallet/pull/2389).

**Pull request closed without merging**